src/components/Navigation.tsx

```diff
@@ -9,30 +9,32 @@ import ServerVersion from "components/ServerVersion";
 import useEventListener from "@use-it/event-listener";
 import { isWidthBelow } from "util/helpers";
 import { useProject } from "context/project";
+import { useMenuCollapsed } from "context/menuCollapsed";
 
 const isSmallScreen = () => isWidthBelow(620);
 
 const Navigation: FC = () => {
-  const [menuCollapsed, setMenuCollapsed] = useState(isSmallScreen());
-
+  const { menuCollapsed, setMenuCollapsed } = useMenuCollapsed();
   const { project, isLoading } = useProject();
 
   const projectName = project && !isLoading ? project.name : "default";
 
   const { isAuthenticated } = useAuth();
   const softToggleMenu = () => {
     if (isSmallScreen()) {
-      setMenuCollapsed((prev) => !prev);
+      setMenuCollapsed(!menuCollapsed);
     }
   };
 
   const hardToggleMenu = (e: MouseEvent<HTMLElement>) => {
-    setMenuCollapsed((prev) => !prev);
+    setMenuCollapsed(!menuCollapsed);
     e.stopPropagation();
   };
 
-  const collapseOnMediumScreen = () => {
-    setMenuCollapsed(isWidthBelow(820));
+  const collapseOnMediumScreen = (e: Event) => {
+    if (!("detail" in e) || e.detail !== "search-and-filter") {
+      setMenuCollapsed(isWidthBelow(820));
+    }
   };
 
   useEventListener("resize", collapseOnMediumScreen);
```

Comment:
I think the event listener and handler should also move into the `menuCollapsed` hook.